task.ps1

```
@@ -6,41 +6,52 @@
 $vnetAddressPrefix = "10.0.0.0/16"
 $subnetAddressPrefix = "10.0.0.0/24"
 $sshKeyName = "linuxboxsshkey"
-$sshKeyPublicKey = Get-Content "~/.ssh/id_rsa.pub" 
-$vmName = "matebox"
+
+# Read SSH key
+$sshKeyPath = Join-Path $HOME ".ssh/id_ed25519.pub"
```

Good: the script reads a local SSH public key and fails early if it's missing. Ensure that the Azure SSH key resource created later (`New-AzSshKey`) is created in the same resource group the validator expects, or remove creation and rely on the existing `linuxboxsshkey`. The validator expects a key resource named `linuxboxsshkey` in the target RG.